---
title: "3. Analysis workflow"
author: Lukas M. Weber
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{3. Analysis workflow}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

Last updated: `r Sys.Date()`


## Visium human DLPFC workflow

In this vignette, we demonstrate the steps in a computational analysis workflow for spatially resolved transcriptomics (ST) data, using an example dataset from the 10x Genomics Visium platform.

As described in the previous vignette, the dataset consists of a single sample (sample 151673) from the human brain dorsolateral prefrontal cortex (DLPFC) region, previously described in our publication [Maynard and Collado-Torres et al. (2021)](https://www.nature.com/articles/s41593-020-00787-0). The full original dataset can be obtained through the [spatialLIBD](http://bioconductor.org/packages/spatialLIBD) Bioconductor package.


## Load data

We begin by loading the dataset. We have stored a copy of this dataset in [SpatialExperiment](https://bioconductor.org/packages/SpatialExperiment) format in the [STexampleData](https://bioconductor.org/packages/STexampleData) package, to make it easier to load for use in examples and tutorials.

If you are using the latest `devel` version of Bioconductor 3.14, the dataset can be loaded as shown in the code below.

Alternatively, if you are working in a `release` version of Bioconductor 3.13, the helper function shown in the code comments can be used instead.

```{r, message=FALSE}
library(SpatialExperiment)
library(STexampleData)

# load dataset from ExperimentHub
spe <- Visium_humanDLPFC()
spe
```

```{r, message=FALSE}
# alternatively: load dataset using helper function if working in Bioc 3.13
# spe <- load_data("Visium_humanDLPFC")
# spe
```


## Plot data

As an initial check, plot the spatial coordinates (spots) in x-y dimensions on the tissue slide, to check that the object has loaded correctly and that the orientation is as expected.

We use visualization functions from the [ggspavis](https://github.com/lmweber/ggpavis) package to generate plots.

```{r, message=FALSE}
library(ggspavis)
```

```{r, fig.height=3.5}
# plot spatial coordinates (spots)
plotSpots(spe)
```


## Quality control (QC)

First, we subset the object to keep only spots over tissue.

```{r}
# subset to keep only spots over tissue
spe <- spe[, spatialData(spe)$in_tissue == 1]
dim(spe)
```

Next, calculate spot-level quality control (QC) metrics using the [scater](https://bioconductor.org/packages/scater) package ([McCarthy et al. 2017](https://doi.org/10.1093/bioinformatics/btw777)) and store the QC metrics in `colData` in the object.

```{r, message=FALSE}
library(scater)
```

```{r}
# identify mitochondrial genes
is_mito <- grepl("(^MT-)|(^mt-)", rowData(spe)$gene_name)
table(is_mito)
rowData(spe)$gene_name[is_mito]

# calculate per-spot QC metrics and store in colData
spe <- addPerCellQC(spe, subsets = list(mito = is_mito))
head(colData(spe), 3)
```

Select filtering thresholds for the QC metrics by examining distributions using histograms. For more details, see the [Quality control](https://lmweber.org/OSTA-book/quality-control.html) chapter in OSTA.

```{r, fig.width=7, fig.height=3}
# histograms of QC metrics
par(mfrow = c(1, 4))
hist(colData(spe)$sum, xlab = "sum", main = "UMIs per spot")
hist(colData(spe)$detected, xlab = "detected", main = "Genes per spot")
hist(colData(spe)$subsets_mito_percent, xlab = "percent mitochondrial", main = "Percent mito UMIs")
hist(colData(spe)$cell_count, xlab = "number of cells", main = "No. cells per spot")
par(mfrow = c(1, 1))

# select QC thresholds
qc_lib_size <- colData(spe)$sum < 500
qc_detected <- colData(spe)$detected < 250
qc_mito <- colData(spe)$subsets_mito_percent > 30
qc_cell_count <- colData(spe)$cell_count > 12

# number of discarded spots for each metric
apply(cbind(qc_lib_size, qc_detected, qc_mito, qc_cell_count), 2, sum)

# combined set of discarded spots
discard <- qc_lib_size | qc_detected | qc_mito | qc_cell_count
table(discard)

# store in object
colData(spe)$discard <- discard
```

Plot the set of discarded spots in x-y coordinates, to confirm that the spatial distribution of the discarded spots does not correspond to any obvious biologically meaningful regions, which would indicate that we are removing biologically informative spots.

```{r, fig.height=3.5}
# check spatial pattern of discarded spots
plotQC(spe, type = "spots", discard = "discard")
```

There is some concentration of discarded spots at the edge of the tissue region, which may be due to tissue damage.

We filter out the low-quality spots from the object.

```{r}
# filter low-quality spots
spe <- spe[, !colData(spe)$discard]
dim(spe)
```


## Normalization

Calculate log-transformed normalized counts, using pool-based size factors and deconvolution to the spot level. We use normalization methods from [scater](https://bioconductor.org/packages/scater) ([McCarthy et al. 2017](https://doi.org/10.1093/bioinformatics/btw777)) and [scran](https://bioconductor.org/packages/scran) ([Lun et al. 2016](https://f1000research.com/articles/5-2122/v2)), by assuming that these methods can be applied by treating spots as equivalent to cells. Since we have a single sample, there are no blocking factors.

```{r, message=FALSE}
library(scran)
```

```{r, fig.height=3.5}
# quick clustering for pool-based size factors
set.seed(123)
qclus <- quickCluster(spe)
table(qclus)

# calculate size factors and store in object
spe <- computeSumFactors(spe, cluster = qclus)

summary(sizeFactors(spe))
hist(sizeFactors(spe), breaks = 20)

# calculate logcounts (log-transformed normalized counts) and store in object
spe <- logNormCounts(spe)

assayNames(spe)
```


## Feature selection

Identify a set of top highly variable genes (HVGs), which will be used to define cell types. We again use methods from [scran](https://bioconductor.org/packages/scran), treating spots as equivalent to cells. We also first filter out mitochondrial genes, since these are very highly expressed and not of main biological interest here.

```{r}
# remove mitochondrial genes
spe <- spe[!is_mito, ]
dim(spe)
```

```{r, message=FALSE, fig.height=5}
# fit mean-variance relationship
dec <- modelGeneVar(spe)

# visualize mean-variance relationship
fit <- metadata(dec)
plot(fit$mean, fit$var, 
     xlab = "mean of log-expression", ylab = "variance of log-expression")
curve(fit$trend(x), col = "dodgerblue", add = TRUE, lwd = 2)

# select top HVGs
top_hvgs <- getTopHVGs(dec, prop = 0.1)
length(top_hvgs)
```


## Dimensionality reduction

Run principal component analysis (PCA) on the set of top HVGs, and retain the top 50 principal components (PCs) for further downstream analyses. This is done to reduce noise and to improve computational efficiency.

We also run UMAP on the set of top 50 PCs and retain the top 2 UMAP components for plotting.

Note that we use the computationally efficient implementation of PCA available in [scater](https://bioconductor.org/packages/scater), which uses randomization, and therefore requires setting a random seed for reproducibility.

```{r}
# compute PCA
set.seed(123)
spe <- runPCA(spe, subset_row = top_hvgs)

reducedDimNames(spe)
dim(reducedDim(spe, "PCA"))
```

```{r}
# compute UMAP on top 50 PCs
set.seed(123)
spe <- runUMAP(spe, dimred = "PCA")

reducedDimNames(spe)
dim(reducedDim(spe, "UMAP"))

# update column names for easier plotting
colnames(reducedDim(spe, "UMAP")) <- paste0("UMAP", 1:2)
```


## Clustering

Next, we perform clustering to define cell types based on molecular expression profiles. We apply graph-based clustering using the Walktrap method implemented in [scran](https://bioconductor.org/packages/scran), applied to the top 50 PCs calculated on the set of top HVGs.

```{r}
# graph-based clustering
set.seed(123)
k <- 10
g <- buildSNNGraph(spe, k = k, use.dimred = "PCA")
g_walk <- igraph::cluster_walktrap(g)
clus <- g_walk$membership
table(clus)

# store cluster labels in column 'label' in colData
colLabels(spe) <- factor(clus)
```

We can visualize the clusters by plotting in (i) x-y coordinates on the tissue slide, and (ii) UMAP dimensions.

From the plots, we can see that the clustering on molecular expression profiles reproduces the known biological structure (cortical layers) to some degree, although not perfectly. The clusters are also separated in UMAP space, but again not perfectly.

```{r, fig.height=3.5}
# plot clusters in x-y coordinates
plotSpots(spe, annotate = "label", 
          palette = "libd_layer_colors")

# plot ground truth labels in x-y coordinates
plotSpots(spe, annotate = "ground_truth", 
          palette = "libd_layer_colors")
```

```{r, fig.width=5, fig.height=4.25}
# plot clusters in UMAP dimensions
plotDimRed(spe, type = "UMAP", 
           annotate = "label", palette = "libd_layer_colors")
```


## Marker genes

Finally, we can identify marker genes by testing for differential gene expression between clusters. We use the `findMarkers` implementation in [scran](https://bioconductor.org/packages/scran), using a binomial test, which tests for genes that differ in the proportion expressed vs. not expressed between clusters. This is a more stringent test than the default t-tests, and tends to select genes that are easier to interpret and validate experimentally.

```{r}
# set gene names as row names for easier plotting
rownames(spe) <- rowData(spe)$gene_name

# test for marker genes
markers <- findMarkers(spe, test = "binom", direction = "up")

# returns a list with one DataFrame per cluster
markers
```

```{r, message=FALSE}
library(pheatmap)
```

```{r, fig.width=5, fig.height=6}
# plot log-fold changes for one cluster over all other clusters
# selecting cluster 1
interesting <- markers[[1]]
best_set <- interesting[interesting$Top <= 5, ]
logFCs <- getMarkerEffects(best_set)

pheatmap(logFCs, breaks = seq(-5, 5, length.out = 101))
```

```{r, fig.width=7, fig.height=7}
# plot log-transformed normalized expression of top genes for one cluster
top_genes <- head(rownames(interesting))

plotExpression(spe, x = "label", features = top_genes)
```

