<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>1. Introduction • OSTAWorkshopBioc2021</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="1. Introduction">
<meta property="og:description" content="OSTAWorkshopBioc2021">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OSTAWorkshopBioc2021</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Vignette01_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/Vignette02_Preprocessing_SPE.html">2. Preprocessing and SpatialExperiment</a>
    </li>
    <li>
      <a href="../articles/Vignette03_Analysis_workflow.html">3. Analysis workflow</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/lmweber/OSTAWorkshopBioc2021">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Vignette01_Introduction_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>1. Introduction</h1>
                        <h4 class="author">Lukas M. Weber</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/lmweber/OSTAWorkshopBioc2021/blob/master/vignettes/Vignette01_Introduction.Rmd"><code>vignettes/Vignette01_Introduction.Rmd</code></a></small>
      <div class="hidden name"><code>Vignette01_Introduction.Rmd</code></div>

    </div>

    
    
<p>Last updated: 2021-08-06</p>
<div id="spatially-resolved-transcriptomics-st" class="section level1">
<h1 class="hasAnchor">
<a href="#spatially-resolved-transcriptomics-st" class="anchor"></a>Spatially resolved transcriptomics (ST)</h1>
<p>Spatially resolved transcriptomics (ST) refers to new technological platforms that allow measuring up to transcriptome-wide gene expression with spatial resolution, e.g. spatial coordinates of the measurements on a tissue slide. Depending on the platform and tissue type, the spatial resolution can be multiple cells per coordinate, approximately single-cell, or sub-cellular.</p>
<p>By combining transcriptome-wide molecular information and spatial information, these platforms can provide powerful new biological insights. For example, we can investigate the spatial distribution of cell types on a tissue sample, colocalization of cell types, or spatial gradients of expression.</p>
<p>Examples of ST platforms include 10x Genomics Visium, Slide-seqV2, seqFISH+, and MERFISH.</p>
<p>Platforms may be grouped into spot-based (transcriptome-wide expression measured at a set of fixed spatial coordinates referred to as spots, e.g. 10x Genomics Visium and Slide-seqV2) and molecule-based (individual RNA molecules and their spatial coordinates are recorded, e.g. seqFISH+ and MERFISH). For more details, see the introductory chapter by Dario Righelli in our <a href="https://drighelli.github.io/SpatialExperiment_Bioc2021/index.html">SpatialExperiment workshop</a>.</p>
</div>
<div id="x-genomics-visium-platform" class="section level1">
<h1 class="hasAnchor">
<a href="#x-genomics-visium-platform" class="anchor"></a>10x Genomics Visium platform</h1>
<p>For this workshop, we use example datasets from the commercially available <a href="https://www.10xgenomics.com/products/spatial-gene-expression">10x Genomics Visium</a> platform.</p>
<p>The following schematic illustrates how this platform works.</p>
<p>This gives us a set of transcriptome-wide expression counts on a grid of approximately 5000 spots, on a capture area of size 6.5mm x 6.5mm.</p>
<div class="figure" style="text-align: center">
<img src="../inst/images/Visium.png" alt="Schematic of 10x Genomics Visium platform. Image source: [10x Genomics Visium website](https://www.10xgenomics.com/spatial-transcriptomics/)" width="100%"><p class="caption">
Schematic of 10x Genomics Visium platform. Image source: <a href="https://www.10xgenomics.com/spatial-transcriptomics/">10x Genomics Visium website</a>
</p>
</div>
</div>
<div id="single-cell-analyses" class="section level1">
<h1 class="hasAnchor">
<a href="#single-cell-analyses" class="anchor"></a>Single-cell analyses</h1>
<p>As an initial approximation, spot-based ST data (e.g. from the Visium platform) can be thought of as similar to single-cell RNA sequencing data, with measurements performed at the level of spots instead of individual cells.</p>
<p>This provides an excellent starting point for analyses, since we have access to a <a href="https://www.scrna-tools.org/">huge number</a> of existing analysis tools for single-cell data, many of them within the Bioconductor framework.</p>
<p>Here, we note that the <a href="https://bioconductor.org/books/release/OSCA/">OSCA</a> book (“Orchestrating Single-Cell Analysis with Bioconductor”) (maintained by Aaron Lun) is an excellent starting point and reference resource if you are interested in analyzing single-cell data with Bioconductor (and was the inspiration for developing OSTA for ST data).</p>
<p>However, this approximation (spots as equivalent to cells) is not perfect – spots may contain zero, one, or many cells – and in addition it ignores any additional information contained in the spatial relationships or distances between spots.</p>
<div class="figure" style="text-align: center">
<img src="../inst/images/OSCA.png" alt="Screenshot of the OSCA online book." width="60%"><p class="caption">
Screenshot of the OSCA online book.
</p>
</div>
</div>
<div id="how-to-analyze-st-data" class="section level1">
<h1 class="hasAnchor">
<a href="#how-to-analyze-st-data" class="anchor"></a>How to analyze ST data?</h1>
<p>This leads us to the question – how best to analyze ST data?</p>
<p>In the <a href="https://lmweber.org/OSTA-book/">OSTA</a> book (“Orchestrating Spatially Resolved Transcriptomics Analysis with Bioconductor”), we aim to describe the steps in a typical computational analysis pipeline for spatially resolved transcriptomics (ST) data, using the Bioconductor framework in R, along with examples of R code and datasets.</p>
<p>We begin by describing preprocessing steps and how to load ST data into R using the <a href="https://bioconductor.org/packages/SpatialExperiment">SpatialExperiment</a> Bioconductor class, followed by downstream analysis steps. For now, we will largely rely on relatively standard methods (e.g. those adapted from single-cell), which have been rigorously evaluated and implemented as Bioconductor packages.</p>
<p>In the future, we expect a growing number of methods will be developed specifically for ST data, which will enable new types of analyses and biological insights. For example, this may include methods for:</p>
<ul>
<li>spatially aware feature selection, e.g. spatially variable genes</li>
<li>spatially aware clustering</li>
<li>deconvolution of cell types in the spatial context</li>
</ul>
<p>Once these new methods have been rigorously tested and implemented within the Bioconductor framework, we will demonstrate them within the OSTA workflows.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Lukas M. Weber.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
