<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>3. Analysis workflow • OSTAWorkshopBioc2021</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="3. Analysis workflow">
<meta property="og:description" content="OSTAWorkshopBioc2021">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OSTAWorkshopBioc2021</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Vignette01_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/Vignette02_Preprocessing_SPE.html">2. Preprocessing and SpatialExperiment</a>
    </li>
    <li>
      <a href="../articles/Vignette03_Analysis_workflow.html">3. Analysis workflow</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/lmweber/OSTAWorkshopBioc2021">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Vignette03_Analysis_workflow_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>3. Analysis workflow</h1>
                        <h4 class="author">Lukas M. Weber</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/lmweber/OSTAWorkshopBioc2021/blob/master/vignettes/Vignette03_Analysis_workflow.Rmd"><code>vignettes/Vignette03_Analysis_workflow.Rmd</code></a></small>
      <div class="hidden name"><code>Vignette03_Analysis_workflow.Rmd</code></div>

    </div>

    
    
<p>Last updated: 2021-08-06</p>
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<div id="visium-human-dlpfc-workflow" class="section level3">
<h3 class="hasAnchor">
<a href="#visium-human-dlpfc-workflow" class="anchor"></a>Visium human DLPFC workflow</h3>
<p>This workflow analyzes one sample of human brain from the dorsolateral prefrontal cortex (DLPFC) region, measured using the 10x Genomics Visium platform.</p>
</div>
<div id="description-of-dataset" class="section level3">
<h3 class="hasAnchor">
<a href="#description-of-dataset" class="anchor"></a>Description of dataset</h3>
<p>This is a 10x Genomics Visium dataset generated from healthy human brain samples from the dorsolateral prefrontal cortex (DLPFC) region.</p>
<p>In the full dataset, there are 12 samples in total, from 3 individuals, with 2 pairs of “spatially adjacent” replicates (serial sections) per individual (4 samples per individual). The individuals and spatially adjacent replicates can be considered as blocking factors. Each sample spans the six layers of the cortex plus white matter in a perpendicular tissue section.</p>
<p>For the examples in this workflow and the analysis chapters, we use a single sample from this dataset (sample 151673).</p>
<p>This dataset is described in our publication <span class="citation">@Maynard2021</span>. The full dataset is publicly available through the <a href="http://bioconductor.org/packages/spatialLIBD">spatialLIBD</a> Bioconductor package, and the analysis code from our paper is provided in the <a href="https://github.com/LieberInstitute/HumanPilot">HumanPilot</a> GitHub repository. The dataset can also be explored interactively through the <a href="http://spatial.libd.org/spatialLIBD/">spatialLIBD Shiny web app</a>.</p>
</div>
<div id="load-data" class="section level3">
<h3 class="hasAnchor">
<a href="#load-data" class="anchor"></a>Load data</h3>
<p>Here, we load a single sample from this dataset (sample 151673), which is available as a <code>SpatialExperiment</code> object from the <a href="https://bioconductor.org/packages/STexampleData">STexampleData</a> package.</p>
<p>For this workshop, we use the latest development version of <a href="https://github.com/lmweber/STexampleData">STexampleData</a> available from GitHub.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">SpatialExperiment</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lmweber/STexampleData">STexampleData</a></span><span class="op">)</span>

<span class="co"># load dataset using latest GitHub version</span>
<span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/STexampleData/man/load_data.html">load_data</a></span><span class="op">(</span><span class="st">"Visium_humanDLPFC"</span><span class="op">)</span>
<span class="va">spe</span>
<span class="co">#&gt; class: SpatialExperiment </span>
<span class="co">#&gt; dim: 33538 4992 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(33538): ENSG00000243485 ENSG00000237613 ... ENSG00000277475</span>
<span class="co">#&gt;   ENSG00000268674</span>
<span class="co">#&gt; rowData names(3): gene_id gene_name feature_type</span>
<span class="co">#&gt; colnames(4992): AAACAACGAATAGTTC-1 AAACAAGTATCTCCCA-1 ...</span>
<span class="co">#&gt;   TTGTTTGTATTACACG-1 TTGTTTGTGTAAATTC-1</span>
<span class="co">#&gt; colData names(3): cell_count ground_truth sample_id</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; mainExpName: NULL</span>
<span class="co">#&gt; altExpNames(0):</span>
<span class="co">#&gt; spatialData names(6) : barcode_id in_tissue ... pxl_col_in_fullres</span>
<span class="co">#&gt;   pxl_row_in_fullres</span>
<span class="co">#&gt; spatialCoords names(2) : x y</span>
<span class="co">#&gt; imgData names(4): sample_id image_id data scaleFactor</span>

<span class="co"># alternatively: load dataset from ExperimentHub</span>
<span class="co"># spe &lt;- Visium_humanDLPFC()</span>
<span class="co"># spe</span></code></pre></div>
</div>
<div id="plot-data" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-data" class="anchor"></a>Plot data</h3>
<p>As an initial check, plot the spatial coordinates (spots) in x-y dimensions on the tissue slide, to check that the object has loaded correctly and that the orientation is as expected.</p>
<p>We use visualization functions from the <a href="https://github.com/lmweber/ggpavis">ggspavis</a> package to generate plots.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lmweber/ggspavis">ggspavis</a></span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># plot spatial coordinates (spots)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggspavis/man/plotSpots.html">plotSpots</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></code></pre></div>
<p><img src="Vignette03_Analysis_workflow_files/figure-html/plot_data-1.png" width="700"></p>
</div>
<div id="quality-control-qc" class="section level3">
<h3 class="hasAnchor">
<a href="#quality-control-qc" class="anchor"></a>Quality control (QC)</h3>
<p>First, we subset the object to keep only spots over tissue.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># subset to keep only spots over tissue</span>
<span class="va">spe</span> <span class="op">&lt;-</span> <span class="va">spe</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-methods.html">spatialData</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">in_tissue</span> <span class="op">==</span> <span class="fl">1</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span>
<span class="co">#&gt; [1] 33538  3639</span></code></pre></div>
<p>Next, calculate spot-level QC metrics using the <code>scater</code> package <span class="citation">[@McCarthy2017]</span>, and store the QC metrics in <code>colData</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/">scater</a></span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># identify mitochondrial genes</span>
<span class="va">is_mito</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"(^MT-)|(^mt-)"</span>, <span class="fu">rowData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">gene_name</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">is_mito</span><span class="op">)</span>
<span class="co">#&gt; is_mito</span>
<span class="co">#&gt; FALSE  TRUE </span>
<span class="co">#&gt; 33525    13</span>
<span class="fu">rowData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">gene_name</span><span class="op">[</span><span class="va">is_mito</span><span class="op">]</span>
<span class="co">#&gt;  [1] "MT-ND1"  "MT-ND2"  "MT-CO1"  "MT-CO2"  "MT-ATP8" "MT-ATP6" "MT-CO3" </span>
<span class="co">#&gt;  [8] "MT-ND3"  "MT-ND4L" "MT-ND4"  "MT-ND5"  "MT-ND6"  "MT-CYB"</span>

<span class="co"># calculate per-spot QC metrics and store in colData</span>
<span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/addPerCellQCMetrics.html">addPerCellQC</a></span><span class="op">(</span><span class="va">spe</span>, subsets <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>mito <span class="op">=</span> <span class="va">is_mito</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-colData.html">colData</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span>, <span class="fl">3</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 3 rows and 9 columns</span>
<span class="co">#&gt;                    cell_count ground_truth     sample_id       sum  detected</span>
<span class="co">#&gt;                     &lt;integer&gt;     &lt;factor&gt;   &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1          6       Layer3 sample_151673      8458      3586</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1         16       Layer1 sample_151673      1667      1150</span>
<span class="co">#&gt; AAACACCAATAACTGC-1          5       WM     sample_151673      3769      1960</span>
<span class="co">#&gt;                    subsets_mito_sum subsets_mito_detected subsets_mito_percent</span>
<span class="co">#&gt;                           &lt;numeric&gt;             &lt;numeric&gt;            &lt;numeric&gt;</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1             1407                    13              16.6351</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1              204                    11              12.2376</span>
<span class="co">#&gt; AAACACCAATAACTGC-1              430                    13              11.4089</span>
<span class="co">#&gt;                        total</span>
<span class="co">#&gt;                    &lt;numeric&gt;</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1      8458</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1      1667</span>
<span class="co">#&gt; AAACACCAATAACTGC-1      3769</span></code></pre></div>
<p>Select filtering thresholds for the QC metrics by examining distributions using histograms.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># histograms of QC metrics</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">4</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-colData.html">colData</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">sum</span>, xlab <span class="op">=</span> <span class="st">"sum"</span>, main <span class="op">=</span> <span class="st">"UMIs per spot"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-colData.html">colData</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">detected</span>, xlab <span class="op">=</span> <span class="st">"detected"</span>, main <span class="op">=</span> <span class="st">"Genes per spot"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-colData.html">colData</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">subsets_mito_percent</span>, xlab <span class="op">=</span> <span class="st">"percent mitochondrial"</span>, main <span class="op">=</span> <span class="st">"Percent mito UMIs"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-colData.html">colData</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">cell_count</span>, xlab <span class="op">=</span> <span class="st">"number of cells"</span>, main <span class="op">=</span> <span class="st">"No. cells per spot"</span><span class="op">)</span></code></pre></div>
<p><img src="Vignette03_Analysis_workflow_files/figure-html/QC_thresholds-1.png" width="672"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span>

<span class="co"># select QC thresholds</span>
<span class="va">qc_lib_size</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-colData.html">colData</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">sum</span> <span class="op">&lt;</span> <span class="fl">500</span>
<span class="va">qc_detected</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-colData.html">colData</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">detected</span> <span class="op">&lt;</span> <span class="fl">250</span>
<span class="va">qc_mito</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-colData.html">colData</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">subsets_mito_percent</span> <span class="op">&gt;</span> <span class="fl">30</span>
<span class="va">qc_cell_count</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-colData.html">colData</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">cell_count</span> <span class="op">&gt;</span> <span class="fl">12</span>

<span class="co"># number of discarded spots for each metric</span>
<span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="va">qc_lib_size</span>, <span class="va">qc_detected</span>, <span class="va">qc_mito</span>, <span class="va">qc_cell_count</span><span class="op">)</span>, <span class="fl">2</span>, <span class="va">sum</span><span class="op">)</span>
<span class="co">#&gt;   qc_lib_size   qc_detected       qc_mito qc_cell_count </span>
<span class="co">#&gt;             7             5             3            47</span>

<span class="co"># combined set of discarded spots</span>
<span class="va">discard</span> <span class="op">&lt;-</span> <span class="va">qc_lib_size</span> <span class="op">|</span> <span class="va">qc_detected</span> <span class="op">|</span> <span class="va">qc_mito</span> <span class="op">|</span> <span class="va">qc_cell_count</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">discard</span><span class="op">)</span>
<span class="co">#&gt; discard</span>
<span class="co">#&gt; FALSE  TRUE </span>
<span class="co">#&gt;  3582    57</span>

<span class="co"># store in object</span>
<span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-colData.html">colData</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">discard</span> <span class="op">&lt;-</span> <span class="va">discard</span></code></pre></div>
<p>Plot the set of discarded spots in the spatial x-y coordinates, to confirm that the spatial distribution of the discarded spots does not correspond to any biologically meaningful regions, which would indicate that we are removing biologically informative spots.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># check spatial pattern of discarded spots</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggspavis/man/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">spe</span>, type <span class="op">=</span> <span class="st">"spots"</span>, discard <span class="op">=</span> <span class="st">"discard"</span><span class="op">)</span></code></pre></div>
<p><img src="Vignette03_Analysis_workflow_files/figure-html/QC_check-1.png" width="700"></p>
<p>There is some concentration of discarded spots at the edge of the tissue region, which may be due to tissue damage.</p>
<p>We filter out the low-quality spots from the object.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># filter low-quality spots</span>
<span class="va">spe</span> <span class="op">&lt;-</span> <span class="va">spe</span><span class="op">[</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-colData.html">colData</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">discard</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span>
<span class="co">#&gt; [1] 33538  3582</span></code></pre></div>
</div>
<div id="normalization" class="section level3">
<h3 class="hasAnchor">
<a href="#normalization" class="anchor"></a>Normalization</h3>
<p>Calculate log-transformed normalized counts, using pool-based size factors and deconvolution to the spot level. We use normalization methods from <code>scater</code> <span class="citation">[@McCarthy2017]</span> and <code>scran</code> <span class="citation">[@Lun2016]</span>, by assuming that these methods can be applied by treating spots as equivalent to cells. Since we have a single sample, there are no blocking factors.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">scran</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># quick clustering for pool-based size factors</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>
<span class="va">qclus</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/quickCluster.html">quickCluster</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">qclus</span><span class="op">)</span>
<span class="co">#&gt; qclus</span>
<span class="co">#&gt;   1   2   3   4   5   6   7   8   9  10 </span>
<span class="co">#&gt; 372 245 254 744 415 230 394 299 492 137</span>

<span class="co"># calculate size factors and store in object</span>
<span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/computeSumFactors.html">computeSumFactors</a></span><span class="op">(</span><span class="va">spe</span>, cluster <span class="op">=</span> <span class="va">qclus</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="fu">sizeFactors</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span>
<span class="co">#&gt;  0.1334  0.6093  0.8844  1.0000  1.2852  4.2475</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="fu">sizeFactors</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></code></pre></div>
<p><img src="Vignette03_Analysis_workflow_files/figure-html/normalization-1.png" width="700"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># calculate logcounts (log-transformed normalized counts) and store in object</span>
<span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html">logNormCounts</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span>

<span class="fu">assayNames</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span>
<span class="co">#&gt; [1] "counts"    "logcounts"</span></code></pre></div>
</div>
<div id="feature-selection" class="section level3">
<h3 class="hasAnchor">
<a href="#feature-selection" class="anchor"></a>Feature selection</h3>
<p>Identify a set of top highly variable genes (HVGs), which will be used to define cell types. We use methods from <code>scran</code> <span class="citation">[@Lun2016]</span>, treating spots as equivalent to cells. We also first filter out mitochondrial genes, since these are very highly expressed and not of biological interest here.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># remove mitochondrial genes</span>
<span class="va">spe</span> <span class="op">&lt;-</span> <span class="va">spe</span><span class="op">[</span><span class="op">!</span><span class="va">is_mito</span>, <span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span>
<span class="co">#&gt; [1] 33525  3582</span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># fit mean-variance relationship</span>
<span class="va">dec</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/modelGeneVar.html">modelGeneVar</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span>

<span class="co"># visualize mean-variance relationship</span>
<span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">metadata</span><span class="op">(</span><span class="va">dec</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">fit</span><span class="op">$</span><span class="va">mean</span>, <span class="va">fit</span><span class="op">$</span><span class="va">var</span>, 
     xlab <span class="op">=</span> <span class="st">"mean of log-expression"</span>, ylab <span class="op">=</span> <span class="st">"variance of log-expression"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/curve.html">curve</a></span><span class="op">(</span><span class="va">fit</span><span class="op">$</span><span class="fu">trend</span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"dodgerblue"</span>, add <span class="op">=</span> <span class="cn">TRUE</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="Vignette03_Analysis_workflow_files/figure-html/feature_selection_HVGs-1.png" width="700"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># select top HVGs</span>
<span class="va">top_hvgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/getTopHVGs.html">getTopHVGs</a></span><span class="op">(</span><span class="va">dec</span>, prop <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">top_hvgs</span><span class="op">)</span>
<span class="co">#&gt; [1] 1448</span></code></pre></div>
</div>
<div id="dimensionality-reduction" class="section level3">
<h3 class="hasAnchor">
<a href="#dimensionality-reduction" class="anchor"></a>Dimensionality reduction</h3>
<p>Run principal component analysis (PCA) on the set of top HVGs, and retain the top 50 principal components (PCs) for further downstream analyses. This is done to reduce noise and to improve computational efficiency. We also run UMAP on the set of top 50 PCs and retain the top 2 UMAP components for visualization purposes.</p>
<p>We use the computationally efficient implementation of PCA available in <code>scater</code> <span class="citation">[@McCarthy2017]</span>, which uses randomization, and therefore requires setting a random seed for reproducibility.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># compute PCA</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>
<span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runPCA.html">runPCA</a></span><span class="op">(</span><span class="va">spe</span>, subset_row <span class="op">=</span> <span class="va">top_hvgs</span><span class="op">)</span>

<span class="fu">reducedDimNames</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span>
<span class="co">#&gt; [1] "PCA"</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="fu">reducedDim</span><span class="op">(</span><span class="va">spe</span>, <span class="st">"PCA"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 3582   50</span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># compute UMAP on top 50 PCs</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>
<span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">spe</span>, dimred <span class="op">=</span> <span class="st">"PCA"</span><span class="op">)</span>

<span class="fu">reducedDimNames</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span>
<span class="co">#&gt; [1] "PCA"  "UMAP"</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="fu">reducedDim</span><span class="op">(</span><span class="va">spe</span>, <span class="st">"UMAP"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 3582    2</span>

<span class="co"># update column names for plotting functions</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="fu">reducedDim</span><span class="op">(</span><span class="va">spe</span>, <span class="st">"UMAP"</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"UMAP"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
</div>
<div id="clustering" class="section level3">
<h3 class="hasAnchor">
<a href="#clustering" class="anchor"></a>Clustering</h3>
<p>Next, we perform clustering to define cell types. We apply graph-based clustering using the Walktrap method implemented in <code>scran</code> <span class="citation">[@Lun2016]</span>, applied to the top 50 PCs calculated on the set of top HVGs.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># graph-based clustering</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>
<span class="va">k</span> <span class="op">&lt;-</span> <span class="fl">10</span>
<span class="va">g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/buildSNNGraph.html">buildSNNGraph</a></span><span class="op">(</span><span class="va">spe</span>, k <span class="op">=</span> <span class="va">k</span>, use.dimred <span class="op">=</span> <span class="st">"PCA"</span><span class="op">)</span>
<span class="va">g_walk</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/cluster_walktrap.html">cluster_walktrap</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span>
<span class="va">clus</span> <span class="op">&lt;-</span> <span class="va">g_walk</span><span class="op">$</span><span class="va">membership</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">clus</span><span class="op">)</span>
<span class="co">#&gt; clus</span>
<span class="co">#&gt;    1    2    3    4    5    6 </span>
<span class="co">#&gt;  372  916  342 1083  349  520</span>

<span class="co"># store cluster labels in column 'label' in colData</span>
<span class="fu">colLabels</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">clus</span><span class="op">)</span></code></pre></div>
<p>Visualize the clusters by plotting in (i) spatial (x-y) coordinates on the tissue slide, and (ii) UMAP dimensions.</p>
<p>From the visualizations, we can see that the clustering reproduces the known biological structure (cortical layers), although not perfectly. The clusters are also separated in UMAP space, but again not perfectly.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># plot clusters in spatial x-y coordinates</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggspavis/man/plotSpots.html">plotSpots</a></span><span class="op">(</span><span class="va">spe</span>, annotate <span class="op">=</span> <span class="st">"label"</span>, 
          palette <span class="op">=</span> <span class="st">"libd_layer_colors"</span><span class="op">)</span></code></pre></div>
<p><img src="Vignette03_Analysis_workflow_files/figure-html/clustering_plots_spots-1.png" width="700"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># plot ground truth labels in spatial coordinates</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggspavis/man/plotSpots.html">plotSpots</a></span><span class="op">(</span><span class="va">spe</span>, annotate <span class="op">=</span> <span class="st">"ground_truth"</span>, 
          palette <span class="op">=</span> <span class="st">"libd_layer_colors"</span><span class="op">)</span></code></pre></div>
<p><img src="Vignette03_Analysis_workflow_files/figure-html/clustering_plots_spots-2.png" width="700"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># plot clusters in UMAP reduced dimensions</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggspavis/man/plotDimRed.html">plotDimRed</a></span><span class="op">(</span><span class="va">spe</span>, type <span class="op">=</span> <span class="st">"UMAP"</span>, 
           annotate <span class="op">=</span> <span class="st">"label"</span>, palette <span class="op">=</span> <span class="st">"libd_layer_colors"</span><span class="op">)</span></code></pre></div>
<p><img src="Vignette03_Analysis_workflow_files/figure-html/clustering_plots_reduced-1.png" width="480"></p>
</div>
<div id="marker-genes" class="section level3">
<h3 class="hasAnchor">
<a href="#marker-genes" class="anchor"></a>Marker genes</h3>
<p>Identify marker genes by testing for differential gene expression between clusters. We use the <code>findMarkers</code> implementation in <code>scran</code> <span class="citation">[@Lun2016]</span>, using a binomial test, which tests for genes that differ in the proportion expressed vs. not expressed between clusters. This is a more stringent test than the default t-tests, and tends to select genes that are easier to interpret and validate experimentally.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># set gene names as row names for easier plotting</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">rowData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">gene_name</span>

<span class="co"># test for marker genes</span>
<span class="va">markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/findMarkers.html">findMarkers</a></span><span class="op">(</span><span class="va">spe</span>, test <span class="op">=</span> <span class="st">"binom"</span>, direction <span class="op">=</span> <span class="st">"up"</span><span class="op">)</span>

<span class="co"># returns a list with one DataFrame per cluster</span>
<span class="va">markers</span>
<span class="co">#&gt; List of length 6</span>
<span class="co">#&gt; names(6): 1 2 3 4 5 6</span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># plot log-fold changes for one cluster over all other clusters</span>
<span class="co"># selecting cluster 1</span>
<span class="va">interesting</span> <span class="op">&lt;-</span> <span class="va">markers</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="va">best_set</span> <span class="op">&lt;-</span> <span class="va">interesting</span><span class="op">[</span><span class="va">interesting</span><span class="op">$</span><span class="va">Top</span> <span class="op">&lt;=</span> <span class="fl">5</span>, <span class="op">]</span>
<span class="va">logFCs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/getMarkerEffects.html">getMarkerEffects</a></span><span class="op">(</span><span class="va">best_set</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap</a></span><span class="op">(</span><span class="va">logFCs</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="op">-</span><span class="fl">5</span>, <span class="fl">5</span>, length.out <span class="op">=</span> <span class="fl">101</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="Vignette03_Analysis_workflow_files/figure-html/marker_genes_heatmap-1.png" width="480"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># plot log-transformed normalized expression of top genes for one cluster</span>
<span class="va">top_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">interesting</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotExpression.html">plotExpression</a></span><span class="op">(</span><span class="va">spe</span>, x <span class="op">=</span> <span class="st">"label"</span>, features <span class="op">=</span> <span class="va">top_genes</span><span class="op">)</span></code></pre></div>
<p><img src="Vignette03_Analysis_workflow_files/figure-html/marker_genes_expression-1.png" width="672"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Lukas M. Weber.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
